webpack/scenes/Hosts/ChangeContentSource/index.js

```diff
@@ -0,0 +1,127 @@
+
+import { translate as __ } from 'foremanReact/common/I18n';
+import { STATUS } from 'foremanReact/constants';
+import { foremanUrl } from 'foremanReact/common/helpers';
+
+import { selectApiDataStatus,
+  selectApiContentViewStatus,
+  selectApiChangeContentStatus,
+  selectContentHostsIds,
+  selectHostsWithoutContent,
+  selectEnvironments,
+  selectContentSources,
+  selectJobInvocationPath,
+  selectContentViews,
+  selectTemplate } from './selectors';
+import { getHostIds, formIsLoading } from './helpers';
+import { getFormData, changeAction, getContentViews } from './actions';
+import ContentSourceForm from './components/ContentSourceForm';
+import ContentSourceTemplate from './components/ContentSourceTemplate';
+import './styles.scss';
+
+const ChangeContentSourcePage = () => {
+  const dispatch = useDispatch();
+
+  const apiDataStatus = useSelector(selectApiDataStatus);
+  const apiContentViewStatus = useSelector(selectApiContentViewStatus);
+  const apiChangeStatus = useSelector(selectApiChangeContentStatus);
+
+  const isLoading = formIsLoading(apiDataStatus, apiContentViewStatus, apiChangeStatus);
+
+  const contentHostsIds = useSelector(selectContentHostsIds);
+  const hostsWithoutContent = useSelector(selectHostsWithoutContent);
+  const environments = useSelector(selectEnvironments);
+  const contentSources = useSelector(selectContentSources);
+  const jobInvocationPath = useSelector(selectJobInvocationPath);
+
+  const template = useSelector(selectTemplate);
+  const contentViews = useSelector(selectContentViews);
+
+  const [environmentId, setEnvironmentId] = useState();
+  const [contentViewId, setContentViewId] = useState();
+  const [contentSourceId, setContentSourceId] = useState();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    dispatch(changeAction(environmentId, contentViewId, contentSourceId, contentHostsIds));
+  };
+
+  const handleEnvironment = (envId) => {
+    if (envId) {
+      dispatch(getContentViews(envId));
+    }
+
+    setEnvironmentId(envId);
+    setContentViewId('');
+  };
+
+  const ignoredHostsAlert = () => (
+    <Alert
+      variant="warning"
+      title={__('Some hosts are ignored!')}
+      className="cs_alert"
+      isExpandable
+    >
+      <p>
+        { __('Following hosts are not managed by Katello, so they will be ignored:') }
+      </p>
+      { hostsWithoutContent.map(name => (
+        <List>
+          <ListItem>
+            <a href={foremanUrl(`/hosts/${name}`)}>{name}</a>
+          </ListItem>
+        </List>))}
+    </Alert>);
+
+  useEffect(() => {
+    dispatch(getFormData());
+  }, [dispatch]);
+
+  if (getHostIds().length === 0) {
+    return (
+      <Grid className="margin-40">
+        <GridItem span={7}>
+          <Alert
+            variant="danger"
+            title={__('No hosts with content source found!')}
+          />
+          { hostsWithoutContent.length > 0 && ignoredHostsAlert() }
```

Comment:
Sometimes I prefer inline JSX rather than defining it above, especially when it makes up most of the component.

```suggestion
          { hostsWithoutContent.length > 0 && 
          <Alert
      variant="warning"
      title={__('Some hosts are ignored!')}
      className="cs_alert"
      isExpandable
    >
      <p>
        { __('Following hosts are not managed by Katello, so they will be ignored:') }
      </p>
      { hostsWithoutContent.map(name => (
        <List>
          <ListItem>
            <a href={foremanUrl(`/hosts/${name}`)}>{name}</a>
          </ListItem>
        </List>))}
    </Alert>
          }
```
Another option is to make `ignoredHostsAlert` a component and render it like 
```js
{ hostsWithoutContent.length > 0 && <IgnoredHostsAlert /> }
```
Either way is fine, but I just really don't like calling regular functions() in order to get JSX :)